Tasks/AzureWebAppV1/deploymentProvider/AzurePublishProfileDeploymentProvider.ts

```diff
@@ -0,0 +1,110 @@
+    }
+
+    public async DeployWebAppStep() {
+        let packageType = this.taskParams.Package.getPackageType();
+        let packagePath = this.taskParams.Package.getPath();
+        await this.kuduServiceUtility.warmpUp();
+        
+        switch(packageType){
+            case PackageType.folder:
+                let tempPackagePath = deployUtility.generateTemporaryFolderOrZipPath(tl.getVariable('AGENT.TEMPDIRECTORY'), false);
+                let archivedWebPackage = await zipUtility.archiveFolder(packagePath, "", tempPackagePath) as string;
+                tl.debug("Compressed folder into zip " +  archivedWebPackage);
+                this.zipDeploymentID = await this.kuduServiceUtility.deployUsingZipDeploy(archivedWebPackage);
+            break;
+
+            case PackageType.zip:
+                this.zipDeploymentID = await this.kuduServiceUtility.deployUsingZipDeploy(packagePath);
+            break;
+
+            case PackageType.jar:
+                tl.debug("Initiated deployment via kudu service for webapp jar package : "+ packagePath);
+                let folderPath = await deployUtility.generateTemporaryFolderForDeployment(false, packagePath, PackageType.jar);
+                let output = await deployUtility.archiveFolderForDeployment(false, folderPath);
+                let webPackage = output.webDeployPkg;
+                tl.debug("Initiated deployment via kudu service for webapp jar package : "+ webPackage);
+                this.zipDeploymentID = await this.kuduServiceUtility.deployUsingZipDeploy(webPackage);
+            break;
+
+            case PackageType.war:
+                tl.debug("Initiated deployment via kudu service for webapp war package : "+ packagePath);
+                let warName = deployUtility.getFileNameFromPath(packagePath, ".war");
+                this.zipDeploymentID = await this.kuduServiceUtility.deployUsingWarDeploy(packagePath, { }, warName);
+            break;
+
+            default:
+                throw new Error('Invalid App Service package or folder path provided: ' + packagePath);
+        }
+    }
+
+    public async UpdateDeploymentStatus(isDeploymentSuccess: boolean) {
+        if(this.kuduServiceUtility) {
+            this.activeDeploymentID = await this.kuduServiceUtility.updateDeploymentStatus(isDeploymentSuccess, null, {'type': 'Deployment'});
+            tl.debug('Active DeploymentId :'+ this.activeDeploymentID);
+
+            if(this.zipDeploymentID && this.activeDeploymentID && isDeploymentSuccess) {
+                await this.kuduServiceUtility.postZipDeployOperation(this.zipDeploymentID, this.activeDeploymentID);
+            }
+        }
+        
+        console.log('App Service Application URL: ' + this.applicationURL);
+        tl.setVariable('AppServiceApplicationUrl', this.applicationURL);
+    }
+
+    private async getSCMCredentialsFromPublishProfile(pubxmlFile: string): Promise<scmCredentials> {
```

Comment:
repeated code. please move to common file

Reply:
I don't want to add another common module dependency as one task uses v1 azure-arm-rest library and other uses v2.

Reply:
we can have a different common module which both the v1 and v2 use.

in worst case, have duplicate code in azure-arm-rest v1 and v2 both. v1 can get cleaned up later leaving only v2. so that way it is better. also this code be user by any other task if it is in common module